packages/forms/src/UIForm/fields/NestedListView/NestedListView.component.js

```diff
@@ -0,0 +1,258 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import keycode from 'keycode';
+import ListView from '@talend/react-components/lib/ListView';
+import { translate } from 'react-i18next';
+
+import { I18N_DOMAIN_FORMS } from '../../../constants';
+import getDefaultT from '../../../translate';
+import { initItems, getDisplayedItems } from './NestedListView.utils';
+import FieldTemplate from '../FieldTemplate';
+
+import theme from './NestedListView.scss';
+
+const DISPLAY_MODE_DEFAULT = 'DISPLAY_MODE_DEFAULT';
+const DISPLAY_MODE_SEARCH = 'DISPLAY_MODE_SEARCH';
+const DEFAULT_ITEM_HEIGHT = 33;
+
+const getItemHeight = () => DEFAULT_ITEM_HEIGHT;
+
+class NestedListViewWidget extends React.Component {
+	constructor(props) {
+		super(props);
+
+		const { t, value, schema, id } = props;
+
+		this.defaultHeaderActions = [
+			{
+				id: `${id}-search`,
+				icon: 'talend-search',
+				label: t('LISTVIEW_WIDGET_SEARCH', { defaultValue: 'Search for specific values' }),
+				onClick: this.switchToSearchMode.bind(this),
+			},
+		];
+
+		this.defaultSearchHeaderActions = [
+			{
+				id: 'abort',
+				icon: 'talend-cross',
+				label: t('LISTVIEW_WIDGET_ABORT', { defaultValue: 'Abort' }),
+				onClick: this.switchToDefaultMode.bind(this),
+			},
+		];
+
+		this.onAddKeyDown = this.onInputKeyDown.bind(this);
+		this.onInputChange = this.onInputChange.bind(this);
+
+		const searchCriteria = null;
+
+		const callbacks = {
+			onExpandToggle: this.onExpandToggle.bind(this),
+			onParentChange: this.onParentChange.bind(this),
+			onCheck: this.onCheck.bind(this),
+		};
+
+		this.state = {
+			...initItems(schema, value, searchCriteria, callbacks),
```

Comment:
Here the items initialized should be stored in the instance, and then the `displayedItems` are calculated based on them for the state.

```javascript
this.items = initItems(schema, callbacks);

this.state = {
   displayedItems: getDisplayedItems(this.items, value, searchCriteria),
   ...
}
```